src/features/Authentication/organisms/SignUpForm/__tests__/SignUpForm.test.js

```diff
@@ -1,31 +1,145 @@
 import React from 'react';
 import {render, screen, waitFor} from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
+import { createClient, Provider } from 'urql';
+import { useNavigate } from 'react-router-dom';
+
 import SignUpForm from 'features/Authentication/organisms/SignUpForm';
+import { server, signUpError } from 'utils/tests';
+import { API_URL } from 'constants/api';
+import { ROUTES } from 'constants/routes';
 
 describe('SignUp Formik form', () => {
-  const dispatch = jest.fn();
-  const renderComponent = () => render(<SignUpForm />);
+  const client = createClient({ url: API_URL });
+
+  const renderComponent = () => render(
+    <Provider value={client}>
+      <SignUpForm />
+    </Provider>
+  );
```

Comment:
We will need to write this again and again in future so it would be better to extract this to the helper